### PR TITLE
xray client: return an error if the HTTP request failed

### PR DESCRIPTION
**Pull request by jaedle**

### CODE REVIEW

File: samplers/aws/xray/internal/client.go
Comment (dmathieu) [inline samplers/aws/xray/internal/client.go:143]:
```suggestion
		return nil, fmt.Errorf("xray client: unable to retrieve sampling settings, expected response status code 200, got: %d", output.StatusCode)
```

File: samplers/aws/xray/internal/client.go
Comment (dmathieu) [inline samplers/aws/xray/internal/client.go:142]: Should this accept any 2xx HTTP responses?

**Review by dmathieu — COMMENTED** [review]
This needs a changelog entry

File: samplers/aws/xray/internal/client.go
Comment (jaedle) [reply samplers/aws/xray/internal/client.go:142]: TBH I asked myself the same question. As the docs only state a successful response should be 200 OK, I would argue that checking for that exact response code should be perfectly fine.
If you think that should be change, feel free to response, so I will change this :)

File: samplers/aws/xray/internal/client.go
Comment (jaedle) [reply samplers/aws/xray/internal/client.go]: done

File: CHANGELOG.md
Comment (dmathieu) [inline CHANGELOG.md:19]:
```suggestion
- Non-200 HTTP status codes when retrieving sampling rules in `go.opentelemetry.io/contrib/samplers/aws/xray/internal/client.go` now return an error. (#5718)
```

**Comment by dmathieu** [comment]
cc @wangzlei as the soon to be owner of this instrumentation.

File: samplers/aws/xray/internal/client_test.go
Comment (dmathieu) [inline samplers/aws/xray/internal/client_test.go:280]: Could you add an assertions checking that the error content matches the proper error? So if somehow, another error is returned, this doesn't still pass.

Comment (jaedle) [reply samplers/aws/xray/internal/client_test.go:280]: Sure, no problem.
I tried to adapt to the existing tests where errors are not explicitly asserted. 

I would recommend to add more specific assertions on all tests related to errors or even add new ones to distinguish error scenarios. 

If that’s fine for you I would work on that.

**Comment by jaedle** [comment]
Thanks for the feedback.

Following you gave a thumbs up on my comment above, I thought that would be appropriate.

> If that’s fine for you, I would work on that within this PR.

Sorry for my confusion.

**Comment by dmathieu** [comment]
My comment was about assertion of errors in tests.
You're also changing the errors content, and changing the way the response body is closed now.

**Comment by jaedle** [comment]
> My comment was about assertion of errors in tests. You're also changing the errors content, and changing the way the response body is closed now.

Rolled back in a moment, thanks.

**Comment by jaedle** [comment]
Sorry for the change, the old way of dealing with response bodies was restored.

**Review by dmathieu — APPROVED** [review]

**Comment by jaedle** [comment]
Hey :wave: 

Any chance to get this merged and released realistically in foreseeable future?

**Comment by dmathieu** [comment]
This needs a second review from a @open-telemetry/go-approvers

**Comment by jaedle** [comment]
Anything I can do from my side to speed things up? It's already 3 month in review.

**Review by dashpole — APPROVED** [review]

File: CHANGELOG.md
Comment (dashpole) [inline CHANGELOG.md:105]: this needs to be moved up to the unreleased section now

**Comment by dashpole** [comment]
Sorry.  We don't have any codeowners for the component, so reviews and such are likely to be much slower.

**Pull request merged**
